stock_orderpoint_generator: add fields trigger and route_id

aayartsev opened the pull request.

aayartsev committed:
stock_orderpoint_generator: add fields trigger and route_id

[FIX] return deleted field

rousseldenis commented on stock_orderpoint_generator/models/orderpoint_template.py:
```diff
@@ -80,6 +80,22 @@ class OrderpointTemplate(models.Model):
+    trigger = fields.Selection(
+        selection=[
+            ("auto", "Auto"),
+            ("manual", "Manual"),
```
@aayartsev Could you explain the aim of this ? As manual triggering for orderpoint creation... is manual creation

aayartsev replied:
@rousseldenis Value of this field will be copy to record wich will be created from this template.

rousseldenis replied:
@aayartsev Could you add your changes in the USAGE.rst ? Thanks

dessanhemrayev replied:
Hi @rousseldenis  )
I have added our changes to USAGE.rst
please check)

geomer198 reviewed (approved):
LGTM!

dessanhemrayev committed:
[IMP] stock_orderpoint_generator: add  new fields

once "Automatic Generation" is enabled, let's display a flag "use product domain": if enabled, a domain for products is displayed, and table "products" is hidden.
With this feature, user does not need to select products one by one, but can apply a domain of products for which a rule should be created automatically.

dessanhemrayev committed:
[IMP] stock_orderpoint_generator: renamed field

dessanhemrayev committed:
[FIX] stock_orderpoint_generator: Moved field "use product domain" from reordering Rule Form to Stock configuration

Before
------

Field "use product domain" managed on the fields "Product domain" and "Product" of the form.

Now
---

Field "use product domain" moved to Stock configuration

- if setting is disabled (standard), behavior remains current one
- if flag is enabled:

in all "reordering rule template" records the "Product" table is not displayed and emptied a "Product domain" field is visible

Add test coverage for this case.
